### PR TITLE
Add default aggregateBy method to RichIterable that takes a target Ma…

### PR DESCRIPTION
**Pull request by donraab**
…p and remove some duplicate implementation code.

Signed-off-by: Donald Raab <Donald.Raab@bnymellon.com>

### CODE REVIEW

File: eclipse-collections-api/src/main/java/org/eclipse/collections/api/RichIterable.java [hunk 1]
```diff
@@ -2422,13 +2422,28 @@ default void appendString(Appendable appendable, String separator)
...
-        MutableMap<K, V> map = Maps.mutable.empty();
+        return this.aggregateBy(
+                groupBy,
+                zeroValueFactory,
+                nonMutatingAggregator,
+                Maps.mutable.empty());
+    }
+
+    /**
+     * @since 10.3
```
Comment (prathasirisha) [inline eclipse-collections-api/src/main/java/org/eclipse/collections/api/RichIterable.java:2433]: One request, can you add a method doc similar to the one above method?

File: eclipse-collections-api/src/main/java/org/eclipse/collections/api/bag/Bag.java [hunk 2]
```diff
@@ -310,4 +313,25 @@ default DoubleSummaryStatistics summarizeDouble(DoubleFunction<? super T> functi
         this.forEachWithOccurrences((each, occurrences) -> target.add(function.valueOf(each, occurrences)));
         return target;
     }
+
+    /**
+     * @since 10.3
+     */
```
Comment (prathasirisha) [inline eclipse-collections-api/src/main/java/org/eclipse/collections/api/bag/Bag.java:319]: same comment as above with doc.

File: eclipse-collections-api/src/main/java/org/eclipse/collections/api/RichIterable.java (hunk 1 above)
Comment (donraab) [reply eclipse-collections-api/src/main/java/org/eclipse/collections/api/RichIterable.java:2437]: Done.

File: eclipse-collections-api/src/main/java/org/eclipse/collections/api/bag/Bag.java (hunk 2 above)
Comment (donraab) [reply eclipse-collections-api/src/main/java/org/eclipse/collections/api/bag/Bag.java:325]: Done.

**Comment by donraab** [comment]
Hey @motlin can you take a look and confirm I got the generics correct on the new aggregateBy methods with target Maps? Thanks!

**Comment by motlin** [comment]
I believe it should be `R extends MutableMapIterable<K, V>` for maximum flexibility. It would be great to show an example that aggregates into a MutableSortedMap.

Very nice, looks like it cuts about 100 lines of code.

**Comment by donraab** [comment]
Good idea on `MutableMapIterable` @motlin!

**Comment by donraab** [comment]
Ok, changed to use `MutableMapIterable`.

**Pull request merged**